check_license/check-headers.sh

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env bash
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# check-headers.sh - check copyright and license in source files
+
+SELF=$0
+
+function usage() {
+	echo "Usage: $SELF <source_root_path> <license_tag> [-h|-v|-a|-d]"
+	echo "   -h, --help          this help message"
+	echo "   -v, --verbose       verbose mode"
+	echo "   -a, --all           check all files (only modified files are checked by default)"
+	echo "   -d, --update_dates  change Copyright dates in all analyzed files (rather not use with -a)"
+}
+
+if [ "$#" -lt 2 ]; then
+	usage >&2
+	exit 2
+fi
+
+SOURCE_ROOT=$1
+shift
+LICENSE=$1
+shift
+
+PATTERN=`mktemp`
+TMP=`mktemp`
+TMP2=`mktemp`
+TEMPFILE=`mktemp`
+rm -f $PATTERN $TMP $TMP2
+
+if [ "$1" == "-h" -o "$1" == "--help" ]; then
+	usage
+	exit 0
+fi
+
+export GIT="git -C ${SOURCE_ROOT}"
+$GIT rev-parse || exit 1
+
+if [ -f $SOURCE_ROOT/.git/shallow ]; then
+	SHALLOW_CLONE=1
+	echo
+	echo "Warning: This is a shallow clone. Checking dates in copyright headers"
+	echo "         will be skipped in case of files that have no history."
+	echo
+else
+	SHALLOW_CLONE=0
+fi
+
+VERBOSE=0
+CHECK_ALL=0
+UPDATE_DATES=0
+while [ "$1" != "" ]; do
+	case $1 in
+	-v|--verbose)
+		VERBOSE=1
+		;;
+	-a|--all)
+		CHECK_ALL=1
+		;;
+	-d|--update_dates)
+		UPDATE_DATES=1
+		;;
+	esac
+	shift
+done
+
+if [ $CHECK_ALL -eq 0 ]; then
+	CURRENT_COMMIT=$($GIT log --pretty=%H -1)
+	MERGE_BASE=$($GIT merge-base HEAD origin/master 2>/dev/null)
+	[ -z $MERGE_BASE ] && \
+		MERGE_BASE=$($GIT log --pretty="%cN:%H" | grep GitHub | head -n1 | cut -d: -f2)
+	[ -z $MERGE_BASE -o "$CURRENT_COMMIT" = "$MERGE_BASE" ] && \
+		CHECK_ALL=1
+fi
+
+if [ $CHECK_ALL -eq 1 ]; then
+	echo "INFO: Checking copyright headers of all files..."
+	GIT_COMMAND="ls-tree -r --name-only HEAD"
+else
+	echo "INFO: Checking copyright headers of modified files only..."
+	GIT_COMMAND="diff --name-only $MERGE_BASE $CURRENT_COMMIT"
+fi
+
+chmod +x "${SOURCE_ROOT}/check_license/file-exceptions.sh"
```

Comment:
`chmod +x` shouldn't be required - you can push the bash file to repository, already with proper file permissions

Reply:
changed